service-base/src/main/java/org/eclipse/hono/service/credentials/SecretsValidator.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+package org.eclipse.hono.service.credentials;
+
+import org.eclipse.hono.service.credentials.validators.AbstractSecretsValidator;
+import org.eclipse.hono.util.CredentialsObject;
+
+/**
+ * Interface that all credentials validators need to implement.
+ * <p>
+ * Any mechanism to authenticate a device is based on credentials secrets that are defined in the
+ * <a href="https://www.eclipse.org/hono/api/Credentials-API/">Credentials API</a>.
+ * This interface defines the methods necessary to provide an algorithm for validating any of these secrets.
+ * <p>
+ * Validators that implement this interface and are implemented as Spring beans in the package
+ * {@link org.eclipse.hono.service.credentials.validators} will be automatically found during startup and are used
+ * during authentication.
+ * This makes adding of own validators very easy.
+ * <p>See the provided validators (e.g. {@link org.eclipse.hono.service.credentials.validators.HashedPasswordValidator})
+ * as a blueprint for how to write own validators.
+ * See {@link AbstractSecretsValidator} as the base class for such
+ * implementations.
+ *
+ * @param <T> The type of what has to be validated (called item below): this can be String in case of password validation, a certificate
+ *           class in case of a client certificate, etc.
+ */
+
+public interface SecretsValidator<T> {
```

Comment:
It verifies only one secret and that is the one provided by the client. It does not matter, that it needs to compare the secret against potentially multiple candidates FMPOV.

BTW The functionality to determine the candidates (based on validity periods) is still missing, isn't it?
So FMPOV it should definitely be `SecretValidator`